packages/utils/src/object.js

```diff
@@ -99,13 +105,38 @@ export const flattenToObject = objectCollection => {
  */
 export const reduceToDictionary = (objectCollection, keyMapper, valueMapper) => {
   const objects = collectionToArray(objectCollection);
-  const getFieldValue = (object, fieldMapper) =>
-    typeof fieldMapper === 'function' ? fieldMapper(object) : object[fieldMapper];
+  const [keyOf, valueOf] = buildKeyAndValueFunctions(keyMapper, valueMapper);
+  const dictionary = {};
+  // Using `forEach` is much quicker than using `reduce` with a spread operator on the accumulator
+  objects.forEach(object => {
+    dictionary[keyOf(object)] = valueOf(object);
+  });
+  return dictionary;
+};
 
+/**
+ * Creates a dictionary which maps the values of a selected field to an array of matching values another field
+ * Use this function when you expect multiple objects in the collection to map to the same key, otherwise use reduceToDictionary
+ * Available field (key/value) mapper types:
+ * * `string`: uses the provided string as the field key
+ * * `Function`, will receive the object as its input eg object => object.value * 2
+ *
+ * @param {ObjectCollection} objectCollection
+ * @param {string|Function} keyMapper
+ * @param {string|Function} valueMapper
+ * @return {Object<string, string[]>}
+ */
+export const reduceToArrayDictionary = (objectCollection, keyMapper, valueMapper) => {
```

Comment:
Added this function for the scenario that we want to use `reduceToDictionary` but we know there will be collisions (ie. matching a parentCode to it's children)